cni/network/multitenancy.go

```diff
@@ -211,7 +211,7 @@ func GetMultiTenancyCNIResult(
 	ifName string) (*cniTypesCurr.Result, *cns.GetNetworkContainerResponse, net.IPNet, *cniTypesCurr.Result, error) {
 
 	if nwCfg.MultiTenancy {
-		result, cnsNetworkConfig, subnetPrefix, err := getContainerNetworkConfiguration(nwCfg, k8sPodName, k8sNamespace, ifName)
+		result, cnsNetworkConfig, subnetPrefix, _, err := getContainerNetworkConfiguration(nwCfg, k8sPodName, k8sNamespace, ifName)
```

Comment:
So here you are basically ignoring the isNotFound flag. Why is that? This function is GetMultiTenancyCNIResult which tries to retrieve the goal state. You are relying only on the err here.
Is the new flag you added really needed? I couldn't understand from the PR as to why exactly we need the new flag and why we cannot use err and cnsNetworkConfig to check if the error occurred during the CNI - CNS call.